api/v1alpha1/akodeploymentconfig_types.go

```diff
@@ -97,26 +97,94 @@ type ExtraConfigs struct {
-	// default value should be antrea, set this string if cluster cni is other type.
-	// enum: calico|canal|flannel|openshift|antrea
+	// default value is antrea, set this string if cluster cni is other type.
+	// AKO supported CNI: antrea|calico|canal|flannel|openshift|ncp
+	// +kubebuilder:validation:Enum=antrea;calico;canal;flannel;openshift;ncp
 	// +optional
 	CniPlugin string `json:"cniPlugin,omitempty"`
+
+	// EnableEVH specifies if you want to enable the Enhanced Virtual Hosting Model
+	// in Avi Controller for the Virtual Services, default value is false
+	// +optional
+	EnableEVH *bool `json:"enableEVH,omitempty"`
+
+	// Layer7Only specifies if you want AKO only to do layer 7 load balancing.
+	// default value is false
+	// +optional
+	Layer7Only *bool `json:"layer7Only,omitempty"`
+
+	// NameSpaceSelector contains label key and value used for namespace migration.
+	// Same label has to be present on namespace/s which needs migration/sync to AKO
+	// +optional
+	NameSpaceSelector NameSpaceSelector `json:"namespaceSelector,omitempty"`
+
+	// ServicesAPI specifies if enables AKO in services API mode: https://kubernetes-sigs.github.io/service-apis/.
+	// Currently, implemented only for L4. This flag uses the upstream GA APIs which are not backward compatible
+	// with the advancedL4 APIs which uses a fork and a version of v1alpha1pre1
+	// default value is false
+	// +optional
+	ServicesAPI *bool `json:"servicesAPI,omitempty"`
+
+	// NetworksConfig specifies the network configurations for virtual services.
+	// +optional
+	NetworksConfig NetworksConfig `json:"networksConfig,omitempty"`
+
+	// IngressConfigs specifies ingress configuration for ako
+	// +optional
+	IngressConfigs AKOIngressConfig `json:"ingress,omitempty"`
+
+	// IngressConfigs specifies L4 load balancer configuration for ako
+	// +optional
+	L4Configs AKOL4Config `json:"l4Config,omitempty"`
+
+	// NodePortSelector only applicable if serviceType is NodePort
+	// +optional
+	NodePortSelector NodePortSelector `json:"nodePortSelector,omitempty"`
+
+	// Rbac specifies the configuration for AKO Rbac
+	// +optional
+	Rbac AKORbacConfig `json:"rbac,omitempty"`
+}
+
+// NameSpaceSelector contains label key and value used for namespace migration
+type NameSpaceSelector struct {
+	LabelKey   string `json:"labelKey,omitempty"`
+	LabelValue string `json:"labelValue,omitempty"`
+}
+
+type NetworksConfig struct {
+	// EnableRHI specifies cluster wide setting for BGP peering.
+	// default value is false
+	// +optional
+	EnableRHI *bool `json:"enableRHI,omitempty"`
+
+	// BGPPeerLabels specifies BGP peers, this is used for selective VsVip advertisement.
+	// +optional
+	BGPPeerLabels []string `json:"bgpPeerLabels,omitempty"`
+
+	// VipNetworkList specifies Network information of the VIP network.
+	// Multiple networks allowed only for AWS Cloud.
+	// default will be the networks specified in Data Networks
+	// +optional
+	// VipNetworkList []NodeNetwork `json:"vipNetworkList,omitempty"`
```

Comment:
I understand that is a required field, and actually, we are filling with `dataNetworkName` and `dataNetworkCIDR` now. 

Didn't expose this is because I saw AKO team is saying:
> Multiple networks allowed only for AWS Cloud.

So currently, I assume that if using AKO with vsphere, we can only set one VIP Network, that is `dataNetworkName` and `dataNetworkCIDR`. 

I have already added this to our interlock meeting agenda with AKO team, we can learn more about this from them.